legate/core/runtime.py

```diff
@@ -883,6 +886,24 @@ def __init__(self, core_library: CoreLib) -> None:
             tuple[int, tuple[ProjExpr, ...]], int
         ] = {}
 
+        self._max_pending_exceptions: int = int(
+            self._core_context.get_tunable(
+                legion.LEGATE_CORE_TUNABLE_MAX_PENDING_EXCEPTIONS,
+                ty.uint32,
+            )
+        )
+        self._precise_exception_trace: bool = (
+            self._max_pending_exceptions > 1
```

Comment:
I guess the point of this condition is that you are going to block directly after the operation is launched anyway (assuming you also flush the scheduling window), so the traceback is going to be almost the same, and thus you can save a stack walk.

Note that in this case the traceback will contain some extraneous frames in the bottom (like the ones you cut with `skip = 2` when calling `capture_stacktrace` elsewhere):

```
Traceback (most recent call last):
  File "/Users/mpapadakis/legate.core/install/lib/python3.8/site-packages/legion_top.py", line 434, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "/Users/mpapadakis/legate.core/install/lib/python3.8/site-packages/legion_top.py", line 255, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "a.py", line 5, in <module>
    cn.linalg.cholesky(cn.random.random((10,10)))
  File "/Users/mpapadakis/cunumeric/cunumeric/coverage.py", line 106, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 88, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/linalg/linalg.py", line 73, in cholesky
    return _cholesky(a)
  File "/Users/mpapadakis/cunumeric/cunumeric/linalg/linalg.py", line 528, in _cholesky
    output._thunk.cholesky(input._thunk, no_tril=no_tril)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 75, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 1869, in cholesky
    cholesky(self, src, no_tril)
  File "/Users/mpapadakis/cunumeric/cunumeric/linalg/cholesky.py", line 183, in cholesky
    potrf_single(context, output.base)
  File "/Users/mpapadakis/cunumeric/cunumeric/linalg/cholesky.py", line 55, in potrf_single
    task.execute()
  File "/Users/mpapadakis/legate.core/install/lib/python3.8/site-packages/legate/core/operation.py", line 167, in execute
    self._context.runtime.submit(self)
  File "/Users/mpapadakis/legate.core/install/lib/python3.8/site-packages/legate/core/runtime.py", line 1040, in submit
    self.raise_exceptions()
  File "/Users/mpapadakis/legate.core/install/lib/python3.8/site-packages/legate/core/runtime.py", line 1383, in raise_exceptions
    pending.raise_exception()
  File "/Users/mpapadakis/legate.core/install/lib/python3.8/site-packages/legate/core/exception.py", line 47, in raise_exception
    raise exn.with_traceback(self._tb)
cunumeric.linalg.exception.LinAlgError: Matrix is not positive definite
```

Since you went to the trouble of throwing such frames away in the "precise" case, you may care to handle this case better as well.

Reply:
8cb62e4764e4bd4921fd2a1b86e848cb888bda32 should remove all legate.core frames.

Reply:
That didn't do anything for me
```
Traceback (most recent call last):
  File "/Users/mpapadakis/legate.core/install/lib/python3.8/site-packages/legion_top.py", line 434, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "/Users/mpapadakis/legate.core/install/lib/python3.8/site-packages/legion_top.py", line 255, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "a.py", line 5, in <module>
    cn.linalg.cholesky(cn.random.random((10,10)))
  File "/Users/mpapadakis/cunumeric/cunumeric/coverage.py", line 106, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 88, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/linalg/linalg.py", line 73, in cholesky
    return _cholesky(a)
  File "/Users/mpapadakis/cunumeric/cunumeric/linalg/linalg.py", line 528, in _cholesky
    output._thunk.cholesky(input._thunk, no_tril=no_tril)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 75, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 1869, in cholesky
    cholesky(self, src, no_tril)
  File "/Users/mpapadakis/cunumeric/cunumeric/linalg/cholesky.py", line 183, in cholesky
    potrf_single(context, output.base)
  File "/Users/mpapadakis/cunumeric/cunumeric/linalg/cholesky.py", line 55, in potrf_single
    task.execute()
  File "/Users/mpapadakis/legate.core/install/lib/python3.8/site-packages/legate/core/operation.py", line 167, in execute
    self._context.runtime.submit(self)
  File "/Users/mpapadakis/legate.core/install/lib/python3.8/site-packages/legate/core/runtime.py", line 1048, in submit
    self.raise_exceptions()
  File "/Users/mpapadakis/legate.core/install/lib/python3.8/site-packages/legate/core/runtime.py", line 1391, in raise_exceptions
    pending.raise_exception()
  File "/Users/mpapadakis/legate.core/install/lib/python3.8/site-packages/legate/core/exception.py", line 47, in raise_exception
    raise exn.with_traceback(self._tb)
cunumeric.linalg.exception.LinAlgError: Matrix is not positive definite
```

Reply:
that's because what you're seeing is not the captured stacktrace, but one directly from Python. I updated the code to apply the precise stacktrace tunable uniformly, so hopefully the stacktrace would be less confusing.